draft-ietf-quic-transport.md

```diff
@@ -1498,31 +1523,42 @@ being used for denial of service.  See {{migrate-validate}} for details.
-Each NEW_CONNECTION_ID is marked with a sequence number.  Connection IDs MUST be
-used in the order in which they are numbered.
-
-A client which wishes to break linkability upon changing networks MUST use the
-connection ID provided by the server as well as incrementing the packet sequence
-number by an externally unpredictable value computed as described in
-{{packet-number-gap}}. Packet number gaps are cumulative.  A client might skip
-connection IDs, but it MUST ensure that it applies the associated packet number
-gaps for connection IDs that it skips in addition to the packet number gap
-associated with the connection ID that it does use.
-
-A server that receives a packet that is marked with a new connection ID recovers
-the packet number by adding the cumulative packet number gap to its expected
-packet number.  A server SHOULD discard packets that contain a smaller gap than
-it advertised.
+to correlate activity between those paths.  An endpoint that moves between
+networks might not wish to have their activity correlated by any entity other
+than a server. The NEW_CONNECTION_ID message can be sent by both endpoints to
+provide an unlinkable connection ID for use in case a peer wishes to explicitly
+break linkability between two points of network attachment.
+
+An endpoint might need to send packets on multiple networks without receiving
+any response from the server.  To ensure that the endpoint is not linkable
+across each of these changes, a new connection ID and packet number gap are
+needed for each network.  To support this, each endpoint sends multiple
+NEW_CONNECTION_ID messages.  Each NEW_CONNECTION_ID is marked with a sequence
+number.  Connection IDs MUST be used in the order in which they are numbered.
+
+An endpoint that does not require the use of a connection ID should not request
+that its peer use a connection ID.  Such an endpoint does not need to provide
+new connection IDs using the NEW_CONNECTION_ID frame.
+
+An endpoint which wishes to break linkability upon changing networks MUST use
+the connection ID provided by its peer as well as incrementing the packet
+sequence number by an externally unpredictable value computed as described in
+{{packet-number-gap}}. Packet number gaps are cumulative.  An endpoint might
+skip connection IDs, but it MUST ensure that it applies the associated packet
+number gaps for connection IDs that it skips in addition to the packet number
+gap associated with the connection ID that it does use.
+
+An endpoint that receives a packet that is marked with a new connection ID
+recovers the packet number by adding the cumulative packet number gap to its
+expected packet number.  An endpoint MUST discard packets that contain a smaller
+gap than it advertised.
+
+An endpoint that receives a packet either with a previously unused connection ID
+or with an empty connection ID and from a new remote address MUST use the next
```

Comment:
If it's an empty connection ID and a new remote address, how would the packet even be associated with the connection?  I think this is a normative requirement for a situation which can't occur.

Reply:
Hmm, only if it does trial decryption for all connections.  Probably unrealistic.  Trimming "with an empty connection ID and" out.

Reply:
Please, drop: "or from a new remote address".  No "either" there.  Otherwise, you are setting yourself up for an attack, where the attacker just replays the same packet from many different addresses toward a peer, draining the list of connection IDs that peer has.  Then, when there is a genuine connection migration, the peer has no connection IDs left.

Reply:
Thanks for updating. The level of control required is not much, really, especially if the connection is over ipv6 (ie. the number of unique ip addresses under attacker's control is enormous).

The ability to collect and correlate data from multiple independent networks to follow nodes across migrations assumes a rather awesome level of control and resources already.